source/module_hamilt_lcao/module_gint/gint_force_cpu_interface.cpp

```diff
@@ -127,44 +191,166 @@ void Gint::cpu_force_meta_interface(Gint_inout* inout) {
+                fvl_dphi_thread);
+                
+            this-> cal_meshball_force(grid_index, na_grid, block_size, block_index,
+                dpsirx_v_DM.get_ptr_2D(), ddpsir_ylm_xx.get_ptr_2D(), ddpsir_ylm_xy.get_ptr_2D(), ddpsir_ylm_xz.get_ptr_2D(), 
+                fvl_dphi_thread);
+            this-> cal_meshball_force(grid_index, na_grid, block_size, block_index,
+                dpsiry_v_DM.get_ptr_2D(), ddpsir_ylm_xy.get_ptr_2D(), ddpsir_ylm_yy.get_ptr_2D(), ddpsir_ylm_yz.get_ptr_2D(), 
+                fvl_dphi_thread);
+            this-> cal_meshball_force(grid_index, na_grid, block_size, block_index,
+                dpsirz_v_DM.get_ptr_2D(), ddpsir_ylm_xz.get_ptr_2D(), ddpsir_ylm_yz.get_ptr_2D(), ddpsir_ylm_zz.get_ptr_2D(), 
+                fvl_dphi_thread);		
+            
+        }
+        if(inout->isstress)
+        {
+            //calculating g_mu(r)*(r-R) where R is the location of atom
+            ModuleBase::Array_Pool<double> array(this->bxyz, LD_pool * 6);
+
+            //the vxc part
+            Gint_Tools::cal_dpsirr_ylm(*this->gridt, this->bxyz, na_grid, grid_index, block_index, block_size, cal_flag,
+                dpsir_ylm_x.get_ptr_2D(), dpsir_ylm_y.get_ptr_2D(),	dpsir_ylm_z.get_ptr_2D(), array.get_ptr_2D());
+            //do integration to get stress
+            this-> cal_meshball_stress(na_grid, block_index, psir_vlbr3_DM.get_ptr_1D(),
+                array.get_ptr_1D(), svl_dphi_thread);
+
+            //partial x of vtau part
+            Gint_Tools::cal_dpsirr_ylm(*this->gridt, this->bxyz, na_grid, grid_index, block_index, block_size, cal_flag,
+                ddpsir_ylm_xx.get_ptr_2D(), ddpsir_ylm_xy.get_ptr_2D(),	ddpsir_ylm_xz.get_ptr_2D(), array.get_ptr_2D());
+            //do integration to get stress
+            this-> cal_meshball_stress(na_grid, block_index, dpsirx_v_DM.get_ptr_1D(),
+                array.get_ptr_1D(), svl_dphi_thread);
+
+            //partial y of vtau part
+            Gint_Tools::cal_dpsirr_ylm(*this->gridt, this->bxyz, na_grid, grid_index, block_index, block_size, cal_flag,
+                ddpsir_ylm_xy.get_ptr_2D(), ddpsir_ylm_yy.get_ptr_2D(),	ddpsir_ylm_yz.get_ptr_2D(), array.get_ptr_2D());
+            //do integration to get stress
+            this-> cal_meshball_stress(na_grid, block_index, dpsiry_v_DM.get_ptr_1D(),
+                array.get_ptr_1D(), svl_dphi_thread);
+
+            //partial z of vtau part
+            Gint_Tools::cal_dpsirr_ylm(*this->gridt, this->bxyz, na_grid, grid_index, block_index, block_size, cal_flag,
+                ddpsir_ylm_xz.get_ptr_2D(), ddpsir_ylm_yz.get_ptr_2D(), ddpsir_ylm_zz.get_ptr_2D(), array.get_ptr_2D());
+            //do integration to get stress
+            this-> cal_meshball_stress(na_grid, block_index, dpsirz_v_DM.get_ptr_1D(),
+                array.get_ptr_1D(), svl_dphi_thread);
+        }
+
+    //release memories
+        delete[] block_iw;
```

Comment:
stop using new and delete

Reply:
The modifications have been completed, including vindex, vldr3, block_iw, block_size, block_index. Meanwhile, cal_flag has been refactored using Array_Pool, and comments in Doxygen format have been added.

Reply:
The modifications have been completed through vector, including vindex, vldr3, block_iw, block_size, block_index. Meanwhile, cal_flag has been refactored using Array_Pool, and comments in Doxygen format have been added.